message_sender/factory.py

```diff
@@ -59,9 +79,60 @@ def _raw_send(self, py_data):
         res = r.json()
         return res.get('result', {})
 
+    def _override_payload(self, payload):
```

Comment:
Since this is quite an involved function, maybe we can add a comment or a docstring just explaining how it works? Something about using `self.override_payload` to transform `payload` into the desired form.